icu4c/source/tools/pkgdata/pkgdata.cpp

```diff
@@ -928,9 +928,8 @@ static void createFileNames(UPKGOptions *o, const char mode, const char *version
         if (IN_DLL_MODE(mode)) {
             sprintf(libFileNames[LIB_FILE], "%s", libName);
         } else {
-            sprintf(libFileNames[LIB_FILE], "%s%s%s",
-                    (strstr(libName, "icudt") ? "lib" : ""),
-                    pkgDataFlags[LIBPREFIX],
+            sprintf(libFileNames[LIB_FILE], "%s%s",
+                    (strstr(libName, "libicudt") ? "" : pkgDataFlags[LIBPREFIX]),
                     libName);
```

Comment:
Also add the same check to the `#else` case, since it fixes the problem for cross-compiling.
`pkgdata --verbose` output:
```
LD_LIBRARY_PATH=/home/KurumiGaming/icu/icu4c/source/build-native/stubdata:/home/KurumiGaming/icu/icu4c/source/build-native/tools/ctestfw:/home/KurumiGaming/icu/icu4c/source/build-native/lib:$LD_LIBRARY_PATH  /home/KurumiGaming/icu/icu4c/source/build-native/bin/pkgdata -O ../data/icupkg.inc -q -c -s /home/KurumiGaming/icu/icu4c/source/build-cross-shared/data/out/build/icudt69l -d ../lib -e icudt69  -T ./out/tmp -p icudt69l -m dll -r 69 -L libicudt  ./out/tmp/icudata.lst -v
# pkgdata: Reading ./out/tmp/icudata.lst..
# Reading options file ../data/icupkg.inc
...
# Writing package file ./out/tmp/icudt69l.dat ..
# libFileName[LIB_FILE] = liblibicudt
# libFileName[LIB_FILE_VERSION] = liblibicudt69.dll
...
```
...which indicates that it uses the `#else` case.
This is just a small part of a more general problem. As it is right now, `pkgdata` is really only suitable for native builds, since the platform-specific bits are being added in using macros, not by checking the host triplet.

Reply:
> This is just a small part of a more general problem. As it is right now, pkgdata is really only suitable for native builds, since the platform-specific bits are being added in using macros, not by checking the host triplet.

Yeah this is a real problem. I build icu with vcpkg, but vcpkg sets `--with-cross-build=` when building mingw on windows because host-triplet=x64-windows and target-triplet=x64-mingw-static. And with `--with-cross-build=` the build fails because of the problem you described

Reply:
We can handle that on vcpkg, by not requiring the host dependency on Windows triplets. But the problem with `pkgdata` not suitable for cross compiling is still there, and needs to be fixed in the future.

Reply:
Yeah I know and I have done that locally. I only wanted to highlight that this is even a problem when you cross build from windows to mingw